.circleci/config.yml

```diff
@@ -183,10 +539,123 @@ jobs:
             else
               echo "Pull Request already exists: $PR_URL"
             fi
-
 workflows:
   version: 2.1
 
+  tests:
+    jobs:
+      - lint
+      - size-contracts
+      - check-packages
+
+      - build-testable
+
+      - test-contracts:
+          name: "test-main"
+          requires: [build-testable]
+          workspace: "@synthetixio/main"
+          parallelism: 2
+
+      - test-contracts:
+          name: "test-oracle-manager"
+          requires: [build-testable]
+          workspace: "@synthetixio/oracle-manager"
+          parallelism: 1
+
+      #- test-contracts: TODO: enable when we have some tests
+      #    name: "test-governance"
+      #    requires: [build-testable]
+      #    workspace: "@synthetixio/governance"
+
+      - test-contracts:
+          name: "test-spot-market"
+          requires: [build-testable]
+          workspace: "@synthetixio/spot-market"
+          parallelism: 1
+
+      - test-contracts:
+          name: "test-perps-market"
+          requires: [build-testable]
+          workspace: "@synthetixio/perps-market"
+          parallelism: 5
+
+      #- test-contracts: TODO: enable when tests are fixed
+      #    name: "test-legacy-market"
+      #    requires: [build-testable]
+      #    workspace: "@synthetixio/legacy-market"
+
+      - test-core-utils:
+          workspace: "@synthetixio/core-utils"
+          parallelism: 2
+
+      - test-core-modules:
+          workspace: "@synthetixio/core-modules"
+          parallelism: 1
+
+      - test-core-contracts:
+          workspace: "@synthetixio/core-contracts"
+          parallelism: 2
+
+      - test-subgraph:
+          name: "test-subgraph-core"
+          workspace: "@synthetixio/core-subgraph"
+
+      - test-subgraph:
+          name: "test-subgraph-spot-market"
+          workspace: "@synthetixio/spot-market-subgraph"
+
+      - test-subgraph:
+          name: "test-subgraph-perps-market"
+          workspace: "@synthetixio/perps-market-subgraph"
+
+      #- simulate-release: TODO: enable when cannon has fixed RPC proxy
```

Comment:
Technically you can start simulating release now for testnets and it's kind of important because there are some invalid release configs that people have tried to do in the past